### PR TITLE
[#88] Create class to retrieve list of trainee's PRs

### PR DESCRIPTION
**Pull request by seyyednavid**

### CODE REVIEW

File: server/app/helpers/ReposService.js [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+    let CYFRepos;
+    try {
+      const getAllPullRequest = getAllCloneRepos;
+      CYFRepos = getAllPullRequest.filter(async (repo) => {
+        const apiUrl = `https://api.github.com/repos/${repo.full_name}`;
+        const response = await fetch(apiUrl);
+        if (response.ok) {
+          const data = await response.json();
+          if (data.parent.full_name.includes("CodeYourFuture")) {
+            // repo.parent = data.parent.full_name;
```
Comment (alexeyschepin) [inline server/app/helpers/ReposService.js:50]: Same here

File: server/app/app.js [hunk 2]
```diff
@@ -55,14 +56,43 @@ const authRoutes = require("./routes/authRoutes");
...
+const ReposService = require("./helpers/ReposService");
 
 app.use("/auth", authRoutes);
 app.use("/signpost", signpostRoute);
 app.use("/user", userRoutes);
 app.use("/current-milestone", milestoneRoute);
-app.use("/codewars", CodewarsRoutes )
+app.use("/codewars", CodewarsRoutes);
+
+app.get("/getAllRepos/:GithubAccount", async (req, res) => {
```
Comment (alexeyschepin) [inline server/app/app.js:68]: Do we need to pass `GithubAccount` param? Can we get it from the authenticated user?

File: server/app/app.js [hunk 3]
```diff
@@ -55,14 +56,43 @@ const authRoutes = require("./routes/authRoutes");
...
+    );
+    res.status(200).json(getAllReposForCYF);
+    const getTrainneReposNumber = await ReposService.getTrainneReposNumber(
+      getAllReposForCYF
+    );
+    // res.status(200).json(getTrainneReposNumber);
+  } catch (error) {
+    console.error(error.message);
+    res.status(500).json({ error: "Failed to fetch pull requests" });
+  }
```
Comment (PaulinaWywrot) [inline server/app/app.js:94]: The error message is slightly misleading, as it seems that we don't fetch the number of pull requests but the number of forked repos.

File: server/app/helpers/ReposService.js [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+      );
+    }
+  }
+
+  static async getAllReposForCYF(getAllCloneRepos) {
+    let CYFRepos;
+    try {
+      const getAllPullRequest = getAllCloneRepos;
+      CYFRepos = getAllPullRequest.filter(async (repo) => {
+        const apiUrl = `https://api.github.com/repos/${repo.full_name}`;
```
Comment (PaulinaWywrot) [inline server/app/helpers/ReposService.js:45]: Again, do we get pull requests data or forked repos data that belong to Code Your Future?

Comment (seyyednavid) [reply server/app/helpers/ReposService.js:45]: Yes

File: server/app/app.js (hunk 2 above)
Comment (seyyednavid) [reply server/app/app.js:68]: I think on the backend we need to get GithubAccount. what do you think?

Comment (alexeyschepin) [reply server/app/app.js:68]: No, the backend already knows trainee's githib username because a trainee must be authenticated to see this page.

**Comment by H-BAHDON** [comment]
this ticket is Blocked by #94 
getting error "exceed the rate limit"

**Review by H-BAHDON — APPROVED** [review]
looks good

**Pull request merged**